src/internal/connector/onedrive/api/drive.go

```diff
@@ -12,6 +13,68 @@ import (
 	"github.com/alcionai/corso/src/internal/connector/graph/api"
 )
 
+func getValues[T any](l api.PageLinker) ([]T, error) {
+	page, ok := l.(interface{ GetValue() []T })
+	if !ok {
+		return nil, errors.Errorf(
+			"response of type [%T] does not comply with GetValue() interface",
+			l,
+		)
+	}
+
+	return page.GetValue(), nil
+}
+
+// max we can do is 999
+const pageSize = int32(999)
+
+type driveItemPager struct {
+	gs      graph.Servicer
+	builder *msdrives.ItemRootDeltaRequestBuilder
+	options *msdrives.ItemRootDeltaRequestBuilderGetRequestConfiguration
+}
+
+func NewItemPager(
+	gs graph.Servicer,
+	driveID, link string,
+	fields []string,
+) *driveItemPager {
+	pageCount := pageSize
+	requestConfig := &msdrives.ItemRootDeltaRequestBuilderGetRequestConfiguration{
+		QueryParameters: &msdrives.ItemRootDeltaRequestBuilderGetQueryParameters{
+			Top:    &pageCount,
+			Select: fields,
+		},
+	}
+
+	res := &driveItemPager{
+		gs:      gs,
+		options: requestConfig,
+		// TODO: Specify a timestamp in the delta query
```

Comment:
Just wondering if this comment is still relevant?

Reply:
Good catch! I think we can remove it. I don't think we want to start with a timestamp for queries based on how the rest of the system is built